packages/idyll-ast/src/index.js

```diff
@@ -58,6 +67,22 @@ const walkNodes = function(ast, f) {
   });
 };
 
+/**
+ * function to do a depth first traversal on the ast tree. 
+ * @param {*} ast     Array that forms the tree structure 
+ * @param {*} f       call-back function 
+ */
+const walkNodesBreadthFirst = function (ast, f) {
+  let childAst = [];
+  //console.log(ast + " length of the ast is: " + ast.length);
+  (ast || []).forEach((node) => {
+    f(node);
+    childAst = childAst.concat(getChildren(node));
+  });
+  if(childAst.length !== 0) {
+      walkNodesBreadthFirst(childAst, f);
```

Comment:
Small knitpick but can you decrease the indentation here (2 spaces), and delete the console.log (rather than commenting it out).